internal/pkg/v2/application/usecases/common/metrics/metrics.go

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+// metrics implements behavior for the metrics use case.
+package metrics
+
+import (
+	dtoBase "github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/dto/v2dot0/common/base"
+	dto "github.com/edgexfoundry/edgex-go/internal/pkg/v2/application/dto/v2dot0/common/metrics"
```

Comment:
These generic alias make the code below harder to understand. Changing to `dtoMetrics` or `metricsDto` makes the following line from below more explicit that the DTO is the a Metrics DTO. 

```
return metricsDto.NewResponse
```
vs 

```
return dto.NewResponse
```
Same goes for `domain` vs `metricsDomain`

I don't have to think about what DTO or Domain it is when reading the code below if the aliases are more explicit.

Reply:
I agree it makes the code more readable.

It also makes the patterns in the code less recognizable.  

I mentioned to you at some point in the last couple of weeks that this was an evolutionary architectural effort.  Which is to say, the design is evolving based on the patterns I detect in the code.  I'm dealing in generics.

On example of this is in the acceptance tests.  A common pattern became readily apparent from the various acceptance tests that were written.  This pattern was extracted within the past week or so into a single implementation (where before it was duplicated across many individual tests) and test.Case was created.

My goal here isn't to make the code more readable.  My goal is to structure the code to make the patterns more recognizable so common functionality can be extracted into its own implementation and the existing individual implementations of that functionality can be imploded.

Meanwhile, my ask is to pay attention to the file you're in (or that's aliased).  

Unresolve and comment if you want to continue this conversation.